src/box/sql/select.c

```diff
@@ -2188,22 +2192,25 @@ sqlColumnsFromExprList(Parse * parse, ExprList * expr_list,
 				if (zName[j] == '_')
 					nName = j;
 			}
-			zName =
-			    sqlMPrintf(db, "%.*z_%u", nName, zName, ++cnt);
+			zName = sqlMPrintf(db, "%.*z_%u", nName, zName, ++cnt);
+			if (zName == NULL)
+				goto cleanup;
 		}
-		size_t name_len = strlen(zName);
+		assert(zName != NULL);
+		nName = sqlStrlen30(zName);
 		void *field = &space_def->fields[i];
-		if (zName != NULL &&
-		    sqlHashInsert(&ht, zName, field) == field)
+		if (sqlHashInsert(&ht, zName, field) == field) {
+			sql_free(zName);
 			sqlOomFault(db);
-		space_def->fields[i].name = region_alloc(region, name_len + 1);
+			goto cleanup;
+		}
+		space_def->fields[i].name = region_alloc(region, nName + 1);
 		if (space_def->fields[i].name == NULL) {
 			sqlOomFault(db);
 			goto cleanup;
```

Comment:
`zName` isn't freed...

Reply:
No, this value is added to a hash table and removed during cleanup below.